overlord/snapstate/handlers.go

```diff
@@ -127,48 +149,47 @@ func (m *SnapManager) doPrerequisites(t *state.Task, _ *tomb.Tomb) error {
 	st.Lock()
 	defer st.Unlock()
 
-	// check if core is installed already
-	_, err := CoreInfo(st)
-	if err == nil {
-		return nil
-	}
-	if err != nil && err != state.ErrNoState {
-		return err
-	}
-
 	// check if we need to inject tasks to install core
 	snapsup, _, err := snapSetupAndState(t)
 	if err != nil {
 		return err
 	}
+
+	// core/ubuntu-core can not have prerequisites
 	snapName := snapsup.Name()
 	if snapName == defaultCoreSnapName || snapName == "ubuntu-core" {
 		return nil
 	}
 
-	// check that there is no task that installs core already
-	for _, chg := range st.Changes() {
-		if chg.Status().Ready() || chg.ID() == t.Change().ID() {
-			continue
-		}
-		for _, tc := range chg.Tasks() {
-			if tc.Kind() == "link-snap" {
-				snapsup, err := TaskSnapSetup(tc)
-				if err != nil {
-					return err
-				}
-				// some other change aleady installs core
-				if snapsup.Name() == defaultCoreSnapName {
-					// if something else installs core
-					// already we need to wait
-					return &state.Retry{After: prerequisitesRetryTimeout}
-				}
-			}
-		}
+	// check prereqs
+	prereqName := defaultCoreSnapName
+	if snapsup.Base != "" {
+		prereqName = snapsup.Base
```

Comment:
Not sure about how/where you want to handle this, but having a base snap doesn't replace the fact we still need a core snap in the system (think re-exec, potentially depended upon by both the base and the snap about to come in).